pkg/operation/botanist/worker.go

```diff
@@ -133,6 +136,7 @@ func (b *Botanist) DeployWorker(ctx context.Context) error {
 			DataVolumes:           dataVolumes,
 			KubeletDataVolumeName: workerPool.KubeletDataVolumeName,
 			Zones:                 workerPool.Zones,
+			AllowSystemComponents: workerPool.AllowSystemComponents,
```

Comment:
I think, we don't need an additional field on the Worker. The node labels are already added via `Worker.spec.pools[].labels` (as you already adapted in line 95). That way, the worker controller doesn't need to act upon this setting and doesn't even need to know about it...

Reply:
Removed